### ISSUE TITLE
[<Firebase Auth>] <Google Sign In>

### ISSUE THREAD

**Issue by TheBossProSniper** [FirebaseExtended/flutterfire#2853]
I am signed in through a google authentication provider in firebase as shown below:
![](https://i.stack.imgur.com/4R6Av.png)

When I tap into the user property of class `FirebaseUser` , when I try to access the providerID, it returns "Firebase" even though I am currently signed in through the Google Sign In provider through Firebase. So, is there any way to find out what provider the user has used for registration of the app? 

Thanks a lot for your help. Below is the code that I have already written.

```
import 'dart:async';
import 'package:flutter/material.dart';
import 'package:lottie/lottie.dart';
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:firebase_auth/firebase_auth.dart';
import 'package:pethub/Components/Sidebar/sidebar_layout_stateful.dart';
import 'package:pethub/View/Screens/main_screen.dart';

FirebaseAuth _auth = FirebaseAuth.instance;
Firestore _firestore = Firestore.instance;
FirebaseUser _user;

class LoadingScreen extends StatefulWidget {
  @override
  _LoadingScreenState createState() => _LoadingScreenState();
}

class _LoadingScreenState extends State<LoadingScreen> {
  int counter = 3;
  Timer timer;

  Future<FirebaseUser> getFirebaseUser() async {
    FirebaseUser user = await _auth.currentUser();
    print(user.email);
    print(user.providerId);
    return user;
  }

  @override
  void initState() {
    super.initState();
    startTimer();
  }

  void startTimer() {
    timer = Timer.periodic(Duration(seconds: 2), (timer) {
      if (getFirebaseUser() != null) {

      }
      Navigator.pushNamed(context, 'welcome');
      timer.cancel();
    });
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        backgroundColor: Colors.black,
        body: Center(
          child: Container(
            color: Colors.black,
            width: 110,
            height: 110,
            child: Lottie.asset('assets/loading.json'),
          ),
        ),
      ),
    );
  }
}
```

I Need to push to the menu screen while providing the parameters so that it can recognise if the user is from Firebase Manual Auth or google sign in provider, which effects the content displayed on the menu screen. 

Here is the problem. This is what it prints :

![](https://i.stack.imgur.com/0Oi0e.png)

**Expected behavior**
Shouldn't it print Google Sign In or something else related to google as the provider? As shown, the user is authenticated in Firebase through google sign in. Is this really supposed to happen? If not can you please let me know how to find out the method through the user has authenticated with Firebase
**Flutter doctor**
```
 tejas@Tejass-Mac-mini  ~/Desktop/PetHub-master   master ●  flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.17.4, on Mac OS X 10.15.5 19F101, locale en-GB)[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
[✓] Android Studio (version 4.0)
[✓] Connected device (1 available)

• No issues found!

```

**Comment by TahaTesser** [comment]
Hi @TheBossProSniper 
You can get this information this way, tested using official code sample
```
    for (var item in user.providerData) {
          print("ProviderData ${item.providerId}");
        }

``` 

**logs**
```
I/flutter ( 7724): ProviderData firebase
I/flutter ( 7724): ProviderData google.com
```

Please try this code sample
```
  void _signInWithGoogle() async {
    final GoogleSignInAccount googleUser = await _googleSignIn.signIn();
    final GoogleSignInAuthentication googleAuth =
        await googleUser.authentication;
    final AuthCredential credential = GoogleAuthProvider.getCredential(
      accessToken: googleAuth.accessToken,
      idToken: googleAuth.idToken,
    );
    final FirebaseUser user =
        (await _auth.signInWithCredential(credential)).user;
    assert(user.email != null);
    assert(user.displayName != null);
    assert(!user.isAnonymous);
    assert(await user.getIdToken() != null);

    final FirebaseUser currentUser = await _auth.currentUser();
    assert(user.uid == currentUser.uid);
    setState(() {
      if (user != null) {
        _success = true;
        _userID = user.uid;
        for (var item in user.providerData) {
          print("ProviderData ${item.providerId}");
        }

      } else {
        _success = false;
      }
    });
  }
```


As you can see, it returns google.com when google sign 
Hope this helps

**Comment by TheBossProSniper** [comment]
It works! Thanks a lot!

**Comment by TahaTesser** [comment]
Hi @TheBossProSniper 
given your last message I feel safe to close this issue, if you disagree please write in the comments and I will reopen it. 
Thank you

**Comment by benedictjohannes** [comment]
I experienced this too.

I created a user account with both email/password and Google sign in. However, the `for` loop @TahaTesser suggested prints `firebase` as the first entry, followwed by `google.com` and `password`.
1. Should `firebase` be listed as `providerId` (is this expected)?
2. When signed in using Google or password' both firebaseUser.providerId returns `firebase` (looks like shortcut to `firebaseUser.providerData[0].providerId`).  
With this, how should the _current_ sign in method be detected? (I need to display this).
